### PR TITLE
Updating template

### PR DESCRIPTION
**Pull request by ScribbleTAS**
- Added common.gradle
- Updated Discombobulator
- Added more properties to gradle.properties
- Updated build.gradle to remove deprecated features

### CODE REVIEW

File: 1.12.2-/common.gradle [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	maven { url = "https://raw.githubusercontent.com/BleachDev/cursed-mappings/main/" }
+}
+
+base { 
+	archivesName = "${ArchiveName}-${ModVersion}"
+}
+
+dependencies {
+	modImplementation "net.fabricmc:fabric-loader:"+project.LoaderVersion
+}
```
Comment (PancakeTAS) [inline 1.12.2-/common.gradle]: just a note for the future, if you're putting something into a block, say for example
```groovy
base {
  archivesName = "a"
}
```
then as long as it's just one line you can simplify it like this:
`base.archivesName = "a"`
no need for it here but I just wanted to mention it

File: 1.12.2-/gradle.properties [hunk 2]
```diff
@@ -1,4 +1,12 @@
...
+#Fabric Settings
+LoaderVersion=0.14.21
+LoomVersion=1.1-SNAPSHOT
+
+#Preprocessor Settings
+DiscombobulatorVersion=1.2.1-SNAPSHOT
+
+#Jar Settings
+ArchiveName=DiscombobulatorTemplate
+ModVersion=1.0.0
```
Comment (PancakeTAS) [inline 1.12.2-/gradle.properties]: I recommend this for naming:
```yaml
# fabric
loader_version=0.14.21
loom_version=1.1-SNAPSHOT

# preprocessor
discombobulator_version=1.2.1-SNAPSHOT

# mod
mod_name=DiscombobulatorTemplate
mod_version=1.0.0
```

File: 1.12.2-/src/main/resources/fabric.mod.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
 	"schemaVersion": 1,
 	"id": "discotemplate",
-	"version": "1.0.0",
+	"version": "${mod_version}",
 	"name": "Test",
 	"description": "A disco template!",
 	"authors": [
```
Comment (PancakeTAS) [inline 1.12.2-/src/main/resources/fabric.mod.json:7]: if we're already making a proper template, then let's also extern all of these properties here. Copy the processResources, gradle.properties and fabric.mod.json from LoTAS or smth

File: 1.14.4+/gradle.properties [hunk 4]
```diff
@@ -1,4 +1,12 @@
...
+#Fabric Settings
+LoaderVersion=0.14.21
+LoomVersion=1.1-SNAPSHOT
+
+#Preprocessor Settings
+DiscombobulatorVersion=1.2.1-SNAPSHOT
+
+#Jar Settings
+ArchiveName=DiscombobulatorTemplate
+ModVersion=1.0.0
```
Comment (PancakeTAS) [inline 1.14.4+/gradle.properties]: same as 1.12.2-

**Review by PancakeTAS — CHANGES_REQUESTED** [review]
small changes

File: 1.12.2-/gradle.properties (hunk 2 above)
Comment (ScribbleTAS) [reply 1.12.2-/gradle.properties]: Great, time to rename every damn thing in every build.gradle

File: 1.12.2-/src/main/resources/fabric.mod.json (hunk 3 above)
Comment (ScribbleTAS) [reply 1.12.2-/src/main/resources/fabric.mod.json:7]: Ima say it. I think putting everything into gradle.properties is unnecessary imo.  
If this stuff is changing a lot during development, like the mod version or maybe the fabric loader dependency, then fine I undestand, but for the most part you set these once and forget about them.

Yes we are making a template here but most of this stuff can't be outsourced to the gradle.properties, without throwing errors in the dev environment, or my IDE screaming at me that there is an error in the json.

You have to open the fabric.mod.json anyway to define the entrypoints, so why not set everything else together with it.

There are also fields in the fabric.mod.json that not everyone needs, like author email. For TASmod I added that for the heck of it and I had to add a field for it in the gradle.properties. Does anyone else use it? Probably not...

**Review by PancakeTAS — APPROVED** [review]

**Pull request merged**